src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -222,30 +220,30 @@ internal void Write(byte[] buf, int offset, int count)
 
         internal void Write(ReadOnlySpan<byte> buf)
         {
-            lock (_fromConnection)
+            lock (_sslContext)
             {
-                foreach (byte b in buf)
-                {
-                    _fromConnection.Enqueue(b);
-                }
+                _inputBuffer.EnsureAvailableSpace(buf.Length);
+                buf.CopyTo(_inputBuffer.AvailableSpan);
+                _inputBuffer.Commit(buf.Length);
             }
         }
 
-        internal int BytesReadyForConnection => _toConnection.Count;
+        internal int BytesReadyForConnection => _outputBuffer.ActiveLength;
 
         internal byte[] ReadPendingWrites()
```

Comment:
How rarely is this called? Allocating every call...

Reply:
It is for example called when handshake is finished to grab any remaining data. To fix that, we would need to rewrite OSX PAL and that is more than I want to do at this point. 
Also based on my measurement, the cost of extra allocation is really small comparing to the cost of the crypto.